components/automate-ui/src/app/modules/infra-proxy/select-box/src/lib/select-box.component.html

```diff
@@ -1,72 +1,149 @@
+            (click)="removeItems()"
+            [disabled]="!getRightSelectedList().length"
+            data-cy="drag-left">
+            <img alt="Right arrow" src="/assets/img/Button.png" />
+          </button>
         </div>
-    </div>
-
-    <div class="vertical empty-box">
-        <div class="select-box-list vertical empty-scroll" cdkDropList #selList="cdkDropList" [cdkDropListConnectedTo]="[origList]"
-            (cdkDropListDropped)="dragOntoRightItems($event)">
-            <div class="select-box-list-item" *ngFor="let item of selectedItems | listFilter:rightFilterText"
-                [ngClass]="item.selected ? 'selected' : ''" (click)="item.selected= (!disabled && !item.selected)"
-                cdkDrag [cdkDragDisabled]="disabled || !item.selected">
-                {{item.value}}
-                <div *cdkDragPlaceholder class="select-box-placeholder vertical">
-                    <span *ngFor="let rightPreview of getRightSelectedList()">{{rightPreview.value}}</span>
-                </div>
-                <div *cdkDragPreview class="select-box-preview vertical">
-                    <span *ngFor="let rightPreview of getRightSelectedList()">{{rightPreview.value}}</span>
+      </div>
+      <div class="vertical empty-box">
+        <div
+          class="vertical empty-scroll"
+          cdkDropList
+          #selList="cdkDropList"
+          [cdkDropListConnectedTo]="[origList]"
+          (cdkDropListDropped)="dragOntoRightItems($event)">
+          <div
+            *ngFor="let item of currentRunList"
+            (click)="item.selected= (!disabled && !item.selected)"
+            cdkDrag
+            [cdkDragDisabled]="disabled || !item.selected"
+            (cdkDragStarted)="onDragrightItemStarted()"
+            class="select-box-list-item"
+            [ngClass]="item.selected ?  'selected' : ''"
+            id="updated-run-list">
+            <div *cdkDragPlaceholder class="select-box-placeholder vertical">
+              <ng-container *ngIf="getRightSelectedList().length > 0 else single_placeholder">
+                <div *ngFor="let rightPreview of getRightSelectedList()" >
+                  <div>{{rightPreview.value}}</div>
                 </div>
+              </ng-container>
+              <ng-template #single_placeholder>
+                <div >{{item.value}}</div>
+              </ng-template>
             </div>
+            {{item.value}}
+          </div>
         </div>
-    </div>
-    <div class="up-down-arrow-box">
+      </div>
+      <div class="up-down-arrow-box">
         <div>
-            <button chef-button class="dropdown-toggle primary" primary (click)="sortItemsInAsccOrder()" [disabled]="disabled">
-                <img alt="Up arrow" src="/assets/img/Button-3.png" />
+          <button
+            chef-button
+            class="dropdown-toggle primary"
+            primary
+            (click)="moveItemUp()"
+            [disabled]="!getRightSelectedList().length"
```

Comment:
I'll double check once this is up and running against the API, but it looks like this is going to fire a continuously running function.  If that is the case, i'll need to be updated to either a pipe or some other sort of variable that only gets checked on changes.

Reply:
yes, i have added a function selectedRunlist in which when we select an item it will get the selected data and will update the button if it is disabled or not.